cli/main.rs

```diff
@@ -693,11 +701,107 @@ async fn test_command(
   global_state
     .file_fetcher
     .save_source_file_in_cache(&main_module, source_file);
+
+  let mut maybe_coverage_collector = if coverage {
+    let inspector = match worker.inspector.as_mut() {
+      Some(inspector) => inspector,
+      None => {
+        return Err(
+          // TODO(caspervonb) this should be implicit when the cover flag is provided.
+          ErrBox::error("coverage option requires --inspect flag"),
+        );
+      }
+    };
+    let inspector_url = Url::parse(&inspector.debugger_url)?;
+    let mut coverage_collector =
+      CoverageCollector::connect(inspector_url).await?;
+    coverage_collector.start_collecting().await?;
+
+    Some(coverage_collector)
+  } else {
+    None
+  };
+
   let execute_result = worker.execute_module(&main_module).await;
   execute_result?;
   worker.execute("window.dispatchEvent(new Event('load'))")?;
   (&mut *worker).await?;
-  worker.execute("window.dispatchEvent(new Event('unload'))")
+  worker.execute("window.dispatchEvent(new Event('unload'))")?;
+  (&mut *worker).await?;
+
+  if let Some(coverage_collector) = maybe_coverage_collector.as_mut() {
+    coverage_collector.stop_collecting().await?;
+
+    // TODO(caspervonb) Use either a thread and join it here or tasks and poll
+    // the worker and coverage_collector concurrently
+    // to avoid having to depend on this somewhat arbritrary delay.
+    tokio::time::delay_for(std::time::Duration::from_millis(500)).await;
+    (&mut *worker).await?;
+
+    let script_coverage = coverage_collector.take_precise_coverage().await?;
+    let test_scripts = test_modules
+      .into_iter()
+      .map(|u| u.to_string())
+      .collect::<Vec<String>>();
+
+    let filtered_coverage = script_coverage
+      .into_iter()
+      .filter(|e| !test_scripts.contains(&e.url))
+      .filter(|e| !e.url.contains(".deno"))
+      .filter(|e| !e.url.contains("__anonymous__"))
+      .filter(|e| e.url.contains(cwd.as_os_str().to_str().unwrap()))
+      .collect::<Vec<ScriptCoverage>>();
```

Comment:
😬  not so great... I think that `cwd` check is not valid - one can run `deno test --coverage ../foo/bar.ts` or `deno test --coverage /tmp/bar.ts` - probably need to integrate with `ModuleGraph` somehow.

Reply:
>  I think that cwd check is not valid - one can run deno test --coverage ../foo/bar.ts

So the idea here was that browsers do blackboxing by domain, since we don't have domains we do it by working directory.

For example; if you cd into let's say std/wasi from deno's root, you don't want std/fs to be included in that coverage report just because it's imported.

Reply:
Sure I get it, but as pointed above you can run files from outside current directory. We should check out how Node tools handle this situation